### PR TITLE
Added plugin for ditaa ReST directive

### PR DESCRIPTION
**Pull request by spookylukey**
Includes automated tests for the happy case, mocking out the `ditaa` binary since that will be hard to install.

I have manually tested the error cases, and could add unit tests for that if wanted.

The [README](https://github.com/spookylukey/plugins/tree/ditaa-plugin/v8/ditaa) should explain the use cases. You can see it in action at https://lukeplant.me.uk/blog/posts/life-on-the-diagonal-adventures-in-2d-time/#a-2-d-calendar

### CODE REVIEW

File: v8/ditaa/ditaa.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+"""Custom reST_ directive for ditaa_ integration.
```
Comment (Kwpolska) [inline v8/ditaa/ditaa.py:1]: Could you add a license header, like the ones seen in Nikola core? (Pick any license you like)

Comment (spookylukey) [reply v8/ditaa/ditaa.py:27]: Sure, added one.

**Review by Kwpolska — APPROVED** [review]

**Comment by Kwpolska** [comment]
Thanks for contributing this!

**Comment by spookylukey** [comment]
Thanks so much for accepting!

**Pull request merged**